### PR TITLE
Extend link element attributes

### PR DESCRIPTION
**Pull request by blfakir**
## Description
Expose link element attributes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/application.js [hunk 1]
```diff
@@ -88,6 +88,23 @@ class XtkSchemaKey {
...
+ * @constructor
+ * @param {} xml
+ * @memberof Campaign
+ */
+ class XtkJoin {
+
+    constructor(xml) {
+        this.src = EntityAccessor.getAttributeAsString(xml, "xpath-src");
+        this.dst = EntityAccessor.getAttributeAsString(xml, "xpath-dst");
+        this.dstFilterExpr = EntityAccessor.getAttributeAsString(xml, "dstFilterExpr");
```
Comment (mkiki) [inline src/application.js:105]: If this is not used, can you remove it?

File: src/application.js [hunk 2]
```diff
@@ -198,6 +225,15 @@ class XtkSchemaNode {
          * @type {string}
          */
         this.nodePath = this._getNodePath(true)._path;
+        /**
+         * Element of type join have a table of join
+         * @type {@type {XtkJoin[]}}
+         */
+        this.join = [];
```
Comment (mkiki) [inline src/application.js:232]: This should be named "joins" as there is more than one join

**Review by mkiki — CHANGES_REQUESTED** [review]

File: src/application.js (hunk 2 above)
Comment (blfakir) [reply src/application.js]: done thx

File: src/application.js (hunk 1 above)
Comment (blfakir) [reply src/application.js]: done

File: src/application.js [hunk 3]
```diff
@@ -173,12 +199,23 @@ class XtkSchemaNode {
...
          * Children of the node. This is a object whose key are the names of the children nodes (without the "@"
          * character for attributes) 
          * @type {Object.<string, Campaign.XtkSchemaNode>}
          */
         this.children = {};
+        /**
+         * Returns an Enumeration object which is the enumeration linked to the current node or null if there is no enumeration
+         * @type {Campaign.XtkEnumeration}
+         */
+        var nodeEnum = EntityAccessor.getAttributeAsString(xml, "enum").split(":")[2];
```
Comment (mkiki) [inline src/application.js:217]: This is unfortunately not correct. Enumerations can reference other schemas

**Review by mkiki — CHANGES_REQUESTED** [review]

File: src/application.js (hunk 3 above)
Comment (blfakir) [reply src/application.js]: change reverted as discussed.

File: src/application.js [hunk 4]
```diff
@@ -204,6 +235,15 @@ class XtkSchemaNode {
          * @type {string}
          */
         this.nodePath = this._getNodePath(true)._path;
+        /**
+         * Element of type join have a table of join
```
Comment (mkiki) [inline src/application.js:239]: Do you mean: Element of type "link" has an array of "join" ?

**Review by mkiki — CHANGES_REQUESTED** [review]
Hello, just a small remark on a comment that seems incorrect and we're good to go.

File: src/application.js (hunk 4 above)
Comment (blfakir) [reply src/application.js]: Yes change made thx

**Review by mkiki — APPROVED** [review]

**Pull request merged**